lib/graphql/union_type.rb

```diff
@@ -64,8 +66,25 @@ def possible_types
       end
     end
 
+    def resolve_type(value, ctx)
+      if !@resolve_type_proc.nil?
+        resolved_type = @resolve_type_proc.call(value, ctx)
+        if !include?(resolved_type)
+          raise(NotImplementedError, "Unrecognized possible type kind: #{resolved_type}")
+        end
+
+        return resolved_type
```

Comment:
It's possible to [hide types and fields](http://graphql-ruby.org/schema/limiting_visibility.html) for individual queries. Should we do anything special here if the user-provided `@resolve_type_proc` returns a type which _ought_ to be hidden for this query?